### DISCUSSION TITLE
Format a code fragment

### DISCUSSION THREAD

**Discussion by JonathanPelah** [zed-industries/zed#11098] [Feature Requests]
Is there a way to apply `edit: format` only on the selected text and not the entire file? [something like Format code fragment on intelij](https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html#reformat_code).

**Comment by SomeoneToIgnore** [comment]
Not really, so far it's either all or nothing, and even the former requires the file to be saved first.

**Reply by JonathanPelah** [reply]
Thanks, any idea for a workaround?

**Reply by SomeoneToIgnore** [reply]
I doubt there's any until it's implemented in Zed, as there's nothing that can, at the same time, expose the selection range outside of Zed and rewrite a part of its file.
I also doubt that this is possible before https://github.com/zed-industries/zed/issues/4529 is implemented.
